### PR TITLE
features_recent_button

### PR DESCRIPTION
**Pull request by lokeshk88**

### CODE REVIEW

**Review by lokeshk88 — COMMENTED** [review]
All done

**Review by lokeshk88 — COMMENTED** [review]
** Add change in food , medication, exercise mvc model.Review it  **

**Comment by joemasilotti** [comment]
Is this ready for review? I can’t tell from the two comments you left - they seem to provide opposite opinions!

**Comment by lokeshk88** [comment]
> Is this ready for review? I can’t tell from the two comments you left - they seem to provide opposite opinions!

yes, is ready but if u need any improper please tell

**Comment by lokeshk88** [comment]
Screenshot from 2024-01-25 13-51-14
Screenshot from 2024-01-25 13-50-51
Screenshot from 2024-01-25 13-50-43
Screenshot from 2024-01-25 13-50-33
Screenshot from 2024-01-25 13-49-34
Screenshot from 2024-01-25 13-46-01

**Review by lokeshk88 — COMMENTED** [review]
**I review its working good.**

**Review by lokeshk88 — COMMENTED** [review]
Check it  @joemasilotti .

File: rails/app/models/exercise_entry.rb [hunk 1]
```diff
@@ -4,4 +4,6 @@ class ExerciseEntry < ApplicationRecord
   belongs_to :user
 
   encrypts :name
+
+  scope :recent, -> { distinct(:name).order(created_at: :desc).limit(3) }
```
Comment (vfonic) [inline rails/app/models/exercise_entry.rb:8]: You have this scope three times. Maybe you can extract it into a concern? `included do scope :recent...`

Concern name could be `Recent`.

File: rails/app/models/water_entry.rb [hunk 2]
```diff
@@ -2,7 +2,7 @@ class WaterEntry < ApplicationRecord
   validates :occurred_on, presence: true
   validates :amount, presence: true
   validates :amount, measured: {greater_than: 0}
-
+  validates_presence_of :amount, :occurred_on
```
Comment (vfonic) [inline rails/app/models/water_entry.rb:5]: This seems redundant. There's already `validates :occurred_on, presence: true` and `validates :amount, presence: true`. This does the same thing, right?

File: rails/app/models/water_entry.rb [hunk 3]
```diff
@@ -17,3 +17,9 @@ def self.units
     }
   end
 end
+
```
Comment (vfonic) [inline rails/app/models/water_entry.rb:20]: Please try to keep the files tidy and not introduce irrelevant changes. You can set up your editor to automatically remove these extra new lines when you save a file.

File: rails/Procfile [hunk 4]
```diff
@@ -1,2 +1,2 @@
-web: bundle exec puma -C config/puma.rb
+web: bundle exec puma -C config/puma.rb -p $PORT
```
Comment (vfonic) [inline rails/Procfile:1]: This change doesn't seem related with the PR.

File: rails/app/controllers/water_entries_controller.rb [hunk 5]
```diff
@@ -16,10 +16,43 @@ def create
     end
   end
 
+  def recent
+    # Assuming 'recent_water_entries' is a method to retrieve recent water entries based on the previous code
+    @recent_water_entries = recent_entries("water")
```
Comment (vfonic) [inline rails/app/controllers/water_entries_controller.rb:21]: Where is this `recent_entries` method declared? I cannot find declaration of this method. This seems to not be fully implemented?

File: rails/app/controllers/water_entries_controller.rb [hunk 6]
```diff
@@ -16,10 +16,43 @@ def create
...
+    params.require(:water_entry).permit(:amount_value, :amount_unit, :occurred_on).merge(user: current_user)
+  end
+
+  def current_day
+    Time.zone.now.to_date
+  end
+
+  def recent_quick_add
+    # Assuming task is passed as a parameter
+    task = params[:amount_unit]
```
Comment (vfonic) [inline rails/app/controllers/water_entries_controller.rb:36]: `task` and `amount_unit` don't seem to depict the same thing. Is this the correct assignment? `task = params[:amount_unit]`

**Review by vfonic — CHANGES_REQUESTED** [review]
Looking at the code, this PR doesn't seem to be fully implemented.

**Pull request closed without merging**